src/controller_plugin.py

```diff
@@ -242,6 +243,65 @@ def get_file(self, file_name, encoding=None):
             return content.decode(encoding)
         return content
 
+    def track_job(self, job):
+        """ Tracks a job by location to prevent request_job_for_client from
+            returning new jobs for locations being tracked. Raises a value
+            error if attempting to insert a job that has a location that is
+            already tracked.
+
+        Arguments:
+            job {dict} -- A job for which its location will be tracked until
+                          respond output is called or the tracking is cleared
+        """
+
+        if self.job_location(job) not in self.tracked_jobs:
+            self.tracked_jobs[self.job_location(job)] = job
+        else:
```

Comment:
Not sure we want to raise an exception here, it would mean that we likely kill the thread every time a duplicate location entry is found.

Reply:
if a job is not found in untrack, the normal way to remove a job from tracking that would be an abnormal situation